src/AutoMoq/Mocking.cs

```diff
@@ -69,12 +69,31 @@ public MockCreationResult CreateAMockObjectFor(Type type, MockBehavior mockBehav
         public MockCreationResult CreateANewMockObjectAndRegisterIt(Type type)
         {
             var result = CreateAMockObjectFor(type);
+
             var mock = (Mock) result.MockObject;
-            ioc.RegisterInstance(mock.Object);
-            ioc.Resolve<AutoMoqer>().SetMock(type, mock);
+
+            RegisterThisObjectInTheIoCContainer(type, mock);
+            RegisterThisMockWithAutoMoq(type, mock);
+
             return result;
         }
 
+        private void RegisterThisMockWithAutoMoq(Type type, Mock mock)
+        {
+            ioc.Resolve<AutoMoqer>().SetMock(type, mock);
+        }
+
+        private void RegisterThisObjectInTheIoCContainer(Type type, Mock mock)
+        {
+            // this is meant to replicate this generic method call
+            // container.RegisterInstance<T>(mock.Object)
+            ioc.GetType()
+                .GetMethods()
+                .First(x => x.Name == "RegisterInstance" && x.IsGenericMethod)
+                .MakeGenericMethod(type)
+                .Invoke(ioc, new[] {mock.Object});
```

Comment:
This was the fix for #29.

After my refactor of "mocking" and "ioc", the registration of the mock objects in the container was switched from ```Register<T>(object)``` to ```Register<object>(object)```.  It didn't seem to affect anything when ```mocker.Create<T>()``` was called once per test, but if it was called twice and the mocks had to be pulled out of the container, they couldn't be retrieved as the mocks were not registered properly.

I needed to get back to ```RegisterInstance<T>```, so I reflected over the container to get that specific method.  That fixed it.